src/commands/moderation/prefixCommands/functions/deleteCommand.ts

```diff
@@ -0,0 +1,105 @@
+import { ChatInputCommandInteraction, Colors, TextChannel, User } from 'discord.js';
+import { constantsConfig, getConn, PrefixCommand, Logger, makeEmbed, clearSinglePrefixCommandCache } from '../../../../lib';
+
+const noConnEmbed = makeEmbed({
+    title: 'Prefix Commands - Delete Command - No Connection',
+    description: 'Could not connect to the database. Unable to delete the prefix command.',
+    color: Colors.Red,
+});
+
+const failedEmbed = (commandId: string) => makeEmbed({
+    title: 'Prefix Commands - Delete Command - Failed',
+    description: `Failed to delete the prefix command with id ${commandId}.`,
+    color: Colors.Red,
+});
+
+const doesNotExistsEmbed = (command: string) => makeEmbed({
+    title: 'Prefix Commands - Delete Command - Does not exist',
+    description: `The prefix command ${command} does not exists. Can not delete it.`,
+    color: Colors.Red,
+});
+
+const successEmbed = (command: string, commandId: string) => makeEmbed({
+    title: `Prefix command ${command} (${commandId}) was deleted successfully.`,
+    color: Colors.Green,
+});
+
+const modLogEmbed = (moderator: User, command: string, aliases: string[], description: string, isEmbed: boolean, embedColor: string, commandId: string) => makeEmbed({
+    title: 'Prefix command deleted',
+    fields: [
+        {
+            name: 'Command',
+            value: command,
+        },
+        {
+            name: 'Moderator',
+            value: `${moderator}`,
+        },
+        {
+            name: 'Aliases',
+            value: aliases.join(','),
+        },
+        {
+            name: 'Description',
+            value: description,
+        },
+        {
+            name: 'Is Embed',
+            value: isEmbed ? 'Yes' : 'No',
+        },
+        {
+            name: 'Embed Color',
+            value: embedColor || '',
+        },
+    ],
+    footer: { text: `Command ID: ${commandId}` },
+    color: Colors.Red,
+});
+
+const noModLogs = makeEmbed({
+    title: 'Prefix Commands - Delete Command - No Mod Log',
+    description: 'I can\'t find the mod logs channel. Please check the channel still exists.',
+    color: Colors.Red,
+});
+
+export async function handleDeletePrefixCommand(interaction: ChatInputCommandInteraction<'cached'>) {
+    await interaction.deferReply({ ephemeral: true });
+
+    const conn = getConn();
+    if (!conn) {
+        await interaction.followUp({ embeds: [noConnEmbed], ephemeral: true });
+        return;
+    }
+
+    const command = interaction.options.getString('command')!;
+    const moderator = interaction.user;
+
+    //Check if the mod logs channel exists
+    const modLogsChannel = interaction.guild.channels.resolve(constantsConfig.channels.MOD_LOGS) as TextChannel;
+    if (!modLogsChannel) {
```

Comment:
```suggestion
    let modLogsChannel = interaction.guild.channels.resolve(constantsConfig.channels.MOD_LOGS);
    if (!modLogsChannel || !modLogsChannel.isTextBased()) {
        modLogsChannel = null;
```
Might be safer to actually check if the channel is a text channel? Especially if we want to offer the codebase to other projects. Then, if it's not a text channel set it to null to prevent accidental attempts to send into a non-text channel.